### DISCUSSION TITLE
Questions from Emacs evil org-mode user

### DISCUSSION THREAD

**Discussion by rieje** [nvim-neorg/neorg#416] [Q&A]
I was wondering if I can get a similar experience using any of the org-mode plugins (the two most promising ones look to be [orgmode](https://github.com/nvim-orgmode/orgmode) (compatible with org files) and [Neorg](https://github.com/nvim-neorg/neorg) (not compatible but seems be a calculated approach for implementing org-mode within the "constraints" of Vim). The features I use most on org-mode (may not necesarily be purely org-mode, but built around it):

* `counsel-org-goto-all`: this will popup a small buffer split horizontally where you can fuzzy match headlines of any depth in all your org files and jump to it. A sub-headline is prefixed with the parent headline to provide the user with more context (useful when headlines may be the same or very similar). `swiper` works the same but search candidates are for each line. If I hold a modifier and move up/down to select the line, the buffer will jump to each candidate so I get a true preview before I decide to commit the jump, closing this selection buffer. With this, it's as if all my notes are in one big org file. Of course there's a companion function which does the same but restricts the candidates to be those in the current file.

* [org-super-agenda](https://github.com/alphapapa/org-super-agenda) provides a customizable org-agenda views where you can group items by criteria, e.g. have a group at the top of the agenda showing items due within 7 days, another that groups items by a certain tag, another that sorts the rest of the items by priority. [org-ql](https://github.com/alphapapa/org-ql) is a compatible tool with org-super-agenda that provides advanced searching and filtering. I would at least like to have features of org-super-agenda for custom agendas that allow for grouping sorted items by todo states, tags, priorities, etc. which allow for a GTD workflow. i find the basic org-agenda to be quite limited, restricted only to time-based views when some things are more long-term and don't necessarily have a concrete date.

* `org-emphasis-alist` allows for styling text. For example, __hello__ might show the text as underlined, /hello/ to italicize, **hello** to show the text in a different color, and +hello+ might add a strikethrough, etc. Of course you can toggle this off and see the actual raw text. Copying the text to the system clipboard can exclude the stylizing portion of the text which is the behavior you usually want.

* Small miscellaneous features like cycling expanding headlines by levels for a very quick and easy way to visualize structure of notes and being able to use bindings like C-s, C-c C-e, C-c c, C-c C-d, C-c C-w, C-M <jk>, etc. I prefer a lot of these bindings to using a leader key followed by 2-character sequence because e.g. C-c C-e means I can hold down C (Ctrl) and roll my finger from c to e as opposed to a sequence of characters like /ce which requires a full keypress on each key. I'm only mentioning this because it seems most people default to using leader keys for all custom bindings and I'm curious if it's overriding essential vim default bindings.

Are there any features that may be difficult to implement in Vim? Much appreciated. I'm not opposed to adopting vimwiki and migrating notes to it, but I feel like it doesn't provide org-agenda-like features of filtering/categorizing/display todo items based on tags, deadlines, and priorities. I'm a little hesitant on neorg because it not being compatible with org files probably means it will break with the fantastic [Orgzly](https://github.com/orgzly/orgzly-android) app, though I respect the fact it seems to be an ambitious project that will overtake the numerous org-mode alternative plugins out there and one actually worth building a workflow around.

Much appreciated.

**Comment by vhyrro** [comment]
Heyo! Two out of four of the features you listed are actually mostly available: `counsel-org-goto-all` and `org-emphasis-alist`.
- Fuzzy finding things within Neorg documents is done through its [telescope](https://github.com/nvim-telescope/telescope.nvim) integration, the repo can be found [here](https://github.com/nvim-neorg/neorg-telescope)!

- Neorg already allows you to style text (it is a markup format, after all). You can see a quick reference guide when you run `:h neorg-basic-markup` in Neovim - hiding these characters is handled natively through nvim's concealing functionality, see `:h conceallevel`/`:h concealcursor` to tweak the exact way you want your styling to be hidden and in what circumstances. The only thing Neorg lacks is excluding the stylized portion while yanking, which we can implement someday as an option.

- Expanding/cycling headlines by levels may be possible, but I can't quite guarantee that. I'll play around with the idea today and see whether Neovim lets me do anything in that regard :p

- Agenda views are part of our bigger GTD implementation which is not complete, nor is it yet stable. GTD features are planned primarily for Neorg 0.2, and the reason GTD is even a thing yet is because @danymat was fantastic enough to start work on it early.

An important thing to note is that Neorg is still alpha! If you're planning to go very serious with your note taking then I don't recommend jumping ship to Neorg just yet, as we're nowhere near as stable as org's ecosystem is - things can change and things can break. Think of using Neorg as being an early adopter. If you go with org as of now we'll have tools in the future to help with migration, so there's no need to make compromises :)

Hope that answers your question! If you need any clarification then ask away.

**Reply by XXiaoA** [reply]
> If you're planning to go very serious with your note taking then I don't recommend jumping ship to Neorg just yet, as we're nowhere near as stable as org's ecosystem is - things can change and things can break.

@vhyrro Hey, this is 2023 now.  I am kinda curious that what would u think now.
BTW, thx for ur great work
